src/main/java/io/matrix/bot/constants/Logger.java

```diff
@@ -0,0 +1,226 @@
+package io.matrix.bot.constants;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.matrix.bot.constants.model.LogData;
+import io.matrix.bot.constants.model.LogLevel;
+import io.matrix.bot.constants.model.LoggerConfig;
+import io.matrix.bot.constants.model.OutputType;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+
+import java.io.IOException;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+import static io.matrix.bot.constants.Util.formatStackTraceString;
+import static io.matrix.bot.constants.accessor.LogLevelAccessor.*;
+import static io.matrix.bot.constants.accessor.MessageAccessor.*;
+
+public class Logger {
+
+    public static final String LOG_LEVELS_JSON_URL = "https://raw.githubusercontent.com/matrixbotio/constants/master/logger/logger.json";
+    private static final String TIME_PLACEHOLDER = "%datetime%";
+    private static final String MESSAGE_PLACEHOLDER = "%message%";
+    private static final String DEFAULT_ERROR_LOG_LEVEL = "error";
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final ExecutorService asyncThreadPool = Executors.newCachedThreadPool();
+    private static final AtomicBoolean close = new AtomicBoolean(false);
+    private static final AtomicInteger numberOfActiveAsyncThreads = new AtomicInteger(0);
+
+    private static LoggerConfig loggerConfig;
+    static {
+        try {
+            loggerConfig = objectMapper.readValue(new URL(LOG_LEVELS_JSON_URL), new TypeReference<>() {});
+        } catch (final IOException e) {
+            printErr(LocalDateTime.now().toString(), "Exception fetching Errors configuration JSON", getErrLogFormat(), e);
```

Comment:
You can move **date** input param inside the method, since it's always **now()**. Same applies to **getErrLogFormat**().

Reply:
For getErrLogFormat - agreed
For date - it's not LocalDateTime that is passed, but it's a String passed, because in not all cases the dateTimeFormat is initialized